solvers/NBody/mobility.h

```diff
@@ -98,6 +106,15 @@ class NBody : public libmobility::Mobility {
 
   void setPositions(device_span<const real> ipositions) override {
     positions.assign(ipositions.begin(), ipositions.end());
+    if (wallHeight != 0)
+    {
+        for (int i = 0; i < numberParticles; i++)
+        {
+            positions[i * 3 + 2] -=
+                wallHeight; // we adjust z so the wall is at 0 since that is how
+                            // the wall kernels are programmed.
+        }
+    }
```

Comment:
This will be super slow, thrust will copy back and forth from the GPU at every access. 
You have to write something like this:
```c++
auto counting_iter = thrust::make_counting_iterator(0);
auto strided_iter = thrust::make_transform_iterator(
    counting_iter,
    thrust::placeholders::_1 * 3
);

auto permZ = thrust::make_permutation_iterator(
    positions.begin() + 2,
    strided_iter
);

thrust::transform(permZ, permZ + numberParticles, permZ,
                  thrust::placeholders::_1 - wallHeight);
```
My solution is really verbose, probably there is a more succint way.